config/spells/moats.json

```diff
@@ -0,0 +1,740 @@
+		"level": 3,
+		"power": 10,
+		"gainChance": {},
+		"levels" : {
+			"base": {
+				"power" : 0,
+				"range" : "0",
+				"description" : "", //For validation
+				"cost" : 0, //For validation
+				"aiValue" : 0, //For validation
+				"battleEffects" : {
+					"directDamage" : {
+						"type":"core:damage"
+					}
+				},
+				"targetModifier":{"smart":false}
+			},
+			"none" : {
+			},
+			"basic" : {
+			},
+			"advanced" : {
+			},
+			"expert" : {
+            }
+		},
+		"flags" : {
+			"damage": true,
+			"negative": true,
+			"nonMagical" : true,
+			"special": true
+		},
+		"targetCondition" : {
+		}
+	},
+    "rampartMoat": {
+        "targetType" : "NO_TARGET",
+        "type": "ability",
+        "name": "Brambles",
+        "school" : {},
+        "level": 1,
+        "power": 10,
+        "defaultGainChance": 0,
+        "gainChance": {},
+        "levels" : {
+            "base":{
+                "description" : "",
+                "aiValue" : 0,
+                "power" : 0,
+                "cost" : 0,
+                "targetModifier":{"smart":false},
+                "battleEffects":{
+                    "moat":{
+                        "type":"core:moat",
+						"hidden" : false,
+						"trap" : true,
+						"trigger" : true,
+						"triggerAbility" : "core:rampartMoatTrigger",
+                        "dispellable" : false,
+						"removeOnTrigger" : false,
+                        "moatDamage" : 70,
+                        "moatHexes" : [[11, 28, 44, 61, 77, 111, 129, 146, 164, 181]],
+						"defender" :{
+						},
+						"bonus" :{
+							"primarySkill" : {
+								"val" : -3,
+								"type" : "PRIMARY_SKILL",
+								"subtype" : "primSkill.defence",
+								"valueType" : "ADDITIVE_VALUE"
+							}
+						}
+                    }
+                },
+                "range" : "X"
+            },
+            "none" :{
+            },
+            "basic" :{
+            },
+            "advanced" :{
+            },
+            "expert" :{
+            }
+        },
+        "flags" : {
+			"nonMagical" : true,
+            "indifferent": true
+        },
+        "targetCondition" : {
+        }
+    },
+    "towerMoat": {
+        "targetType" : "NO_TARGET",
+        "type": "ability",
+        "name": "Land Mine",
+        "school" : {},
+        "level": 3,
+        "power": 10,
+        "defaultGainChance": 0,
+        "gainChance": {},
+        "levels" : {
+            "base":{
+                "description" : "",
+                "aiValue" : 0,
+                "power" : 0,
+                "cost" : 0,
+                "targetModifier":{"smart":false},
+                "battleEffects":{
+                    "moat":{
+                        "type":"core:moat",
+						"hidden" : true,
+						"trap" : false,
+						"trigger" : true,
+						"triggerAbility" : "core:landMineTrigger",
+                        "dispellable" : true,
+						"removeOnTrigger" : true,
+                        "moatDamage" : 150,
```

Comment:
Will moat damage work correctly if town is guarded by strong hero who can cast Mines with damage greater than 150?
See "Tower's Land Mines" section here: https://heroes.thelazy.net//index.php/Moat

Reply:
Yes, and it is a purpose of previous check for hero to have a caster. So, if hero is present, mines will have more than 150 damage if caster can have more effect power and effect level. You cannot have less than moatDamage, but can have more. Only one case which I did not check - how sorcery of defender actually affects tower's landmines in OH3.

Reply:
In this PR this always affects damage, so, level 1 Solmyr will have 157 mines damage.

Reply:
FizMig says than sorcery is affect mines like spell, so, I think it is safe to say than we work most likely correct.

Reply:
test in original H3 with solmyr?

Reply:
@dydzio0614 I do not know a cheat to force build all buildings for AI in OH3. Do you know some?

Reply:
it's called map editor xD

Reply:
@dydzio0614 in OH3 sorcery of level 1 solmyr also gives mines 157 dmg) so, I did correctly.